### PR TITLE
Make system id mandatory in items #157

### LINKED ISSUE #157: Make system ID mandatory in item
The `system_id` is optional/nullable when creating an item but this should be changed so that it is mandatory.

### PR DESCRIPTION
**Pull request by joelvdavies**
## Description
Makes system id mandatory in items. Front end changes have already been merged.

Also cleans up some inconsistencies I saw in the e2e tests for items.

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage

## Agile board tracking
Closes #157

### CODE REVIEW

File: inventory_management_system_api/schemas/item.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ class ItemPostRequestSchema(BaseModel):
     """
 
     catalogue_item_id: str = Field(description="The ID of the corresponding catalogue item for this item")
-    system_id: Optional[str] = Field(default=None, description="The ID of the system that the item belongs to")
+    system_id: str = Field(default=None, description="The ID of the system that the item belongs to")
```
Comment (VKTB) [inline inventory_management_system_api/schemas/item.py:30]: Can `default=None` be removed now?
```suggestion
    system_id: str = Field(description="The ID of the system that the item belongs to")
```

File: inventory_management_system_api/repositories/item.py [hunk 2]
```diff
@@ -86,7 +86,7 @@ def list(self, system_id: Optional[str], catalogue_item_id: Optional[str]) -> Li
         query = {}
         if system_id:
             # allows users to filter by items with no system ID
```
Comment (VKTB) [inline inventory_management_system_api/repositories/item.py:88]: Can this comment be removed now?
```suggestion
```

File: inventory_management_system_api/schemas/item.py [hunk 3]
```diff
@@ -53,6 +53,7 @@ class ItemPatchRequestSchema(ItemPostRequestSchema):
     catalogue_item_id: Optional[str] = Field(
         default=None, description="The ID of the corresponding catalogue item for this item"
     )
+    system_id: Optional[str] = Field(default=None, description="The ID of the system that the item belongs to")
```
Comment (VKTB) [inline inventory_management_system_api/schemas/item.py:56]: Can `default=None` be removed now?
```suggestion
    system_id: Optional[str] = Field(description="The ID of the system that the item belongs to")
```

Comment (joelvdavies) [reply inventory_management_system_api/schemas/item.py:56]: This one is patch so it should be optional.

**Review by VKTB — APPROVED** [review]

**Pull request merged**